### ISSUE TITLE
Release APKs?

### ISSUE THREAD

**Issue by IzzySoft** [cmeng-git/xmail#3]
Would you consider providing signed APKs of release builds here (ideally attached to their corresponding [Releases](https://github.com/cmeng-git/xmail/releases)? That would enable those outside of Google's walled garden to use your app. Thanks in advance!

**Comment by cmeng-git** [comment]
The playstore release version does not have the auto software update service enabled/included. It depends on google playstore to support for any new release update.

The debug version enable the auto software update service, and it will alert user if there are any new software release.

Please note that the playstore version is signed by google; any own self release version signature is different, hence cannot install over each other.

Any comment?

**Comment by IzzySoft** [comment]
> The playstore release version does not have the auto software update service enabled/included. It depends on google playstore to support for any new release update.

Oh, I can relieve you of that problem: My intention was to see that I can provide your app via [my F-Droid repo](https://apt.izzysoft.de/fdroid), which then would cover that (my updater checker runs once a day). Maybe you remember me and my repo from the past? Didn't I carry aTalk until it finally got listed by F-Droid? Of course it would be great could we get xmail to F-Droid.org in the end, too :smiley:

> Please note that the playstore version is signed by google; any own self release version signature is different, hence cannot install over each other.

So Google is the sole party with access to your signing key? Well… I guess those getting their apps from F-Droid would continue getting them from F-Droid. If the signature is not identical as you have no access to that signing key, that cannot be helped (though of course ideally the two would match). Plus unless you achieve reproducible builds, F-Droid would sign their builds with a different key as well, as is the case with aTalk.

**TL;DR:** For those without access to PlayStore, it would still be better to have an APK signed with a different key than none at all, right? So if these two are the only issues you see, I hope we can consider them dispelled?

And PS: I gladly help you getting xmail to F-Droid.org then if you wish.

**Comment by cmeng-git** [comment]
> Oh, I can relieve you of that problem: My intention was to see that I can provide your app via [my F-Droid repo](https://apt.izzysoft.de/fdroid),

OK, I can place a copy of the xmail in the release directory, then you can put it into your repo.

> And PS: I gladly help you getting xmail to F-Droid.org then if you wish.

Actually I do not spent much resource on this app for any further enhancements, except to upgrade it so it is compatible with the latest android-API.  I am currently trying to port aTalk to harmonyOS, and this will consume all my resource time. I am afraid I do not have extra capacity to do further development on xmail. May be it is no appropriate to release into Fdroid.

**Comment by IzzySoft** [comment]
> OK, I can place a copy of the xmail in the release directory, then you can put it into your repo.

Thanks! I'll then wait for your closing this issue as signal the APK is ready for pick-up (and hence leave it open for now).

> I am afraid I do not have extra capacity to do further development on xmail.

Understood. Should that change one day, my offer stands :wink: Meanwhile, all the best for your "migration task"!

**Comment by cmeng-git** [comment]
Release apk uploaded.

**Comment by IzzySoft** [comment]
Thanks! Becomes [available here](https://apt.izzysoft.de/packages/org.atalk.xryptomail/) with the next sync around 6 pm UTC then. Be welcome to [pick a badge](https://gitlab.com/IzzyOnDroid/repo/#if-my-app-is-available-here-do-you-have-a-badge-i-can-use-to-link-to-it) to link there e.g. from the Readme if you wish :smiley:
